components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,848 @@
+        month_code: types::MonthCode,
+        day: u8,
+    ) -> Result<Self::DateInner, CalendarError> {
+        let year = if era.0 == tinystr!(16, "hebrew") {
+            year
+        } else {
+            return Err(CalendarError::UnknownEra(era.0, self.debug_name()));
+        };
+
+        ArithmeticDate::new_from_codes(self, year, month_code, day).map(HebrewDateInner)
+    }
+
+    fn date_from_iso(&self, iso: Date<Iso>) -> Self::DateInner {
+        let fixed_iso = Iso::fixed_from_iso(*iso.inner());
+        Self::hebrew_from_fixed(fixed_iso).inner
+    }
+
+    fn date_to_iso(&self, date: &Self::DateInner) -> Date<Iso> {
+        let fixed_hebrew = Self::fixed_from_hebrew(*date);
+        Iso::iso_from_fixed(fixed_hebrew)
+    }
+
+    fn months_in_year(&self, date: &Self::DateInner) -> u8 {
+        date.0.months_in_year()
+    }
+
+    fn days_in_year(&self, date: &Self::DateInner) -> u16 {
+        date.0.days_in_year()
+    }
+
+    fn days_in_month(&self, date: &Self::DateInner) -> u8 {
+        date.0.days_in_month()
+    }
+
+    fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
+        date.0.offset_date(offset)
+    }
+
+    fn until(
+        &self,
+        date1: &Self::DateInner,
+        date2: &Self::DateInner,
+        _calendar2: &Self,
+        _largest_unit: DateDurationUnit,
+        _smallest_unit: DateDurationUnit,
+    ) -> DateDuration<Self> {
+        date1.0.until(date2.0, _largest_unit, _smallest_unit)
+    }
+
+    fn debug_name(&self) -> &'static str {
+        "Hebrew"
+    }
+
+    fn year(&self, date: &Self::DateInner) -> types::FormattableYear {
+        Self::year_as_hebrew(date.0.year)
+    }
+
+    fn month(&self, date: &Self::DateInner) -> types::FormattableMonth {
+        date.0.month()
+    }
+
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        date.0.day_of_month()
+    }
+
+    fn day_of_year_info(&self, date: &Self::DateInner) -> types::DayOfYearInfo {
+        let prev_year = date.0.year.saturating_sub(1);
+        let next_year = date.0.year.saturating_add(1);
+        types::DayOfYearInfo {
+            day_of_year: date.0.day_of_year(),
+            days_in_year: date.0.days_in_year(),
+            prev_year: Self::year_as_hebrew(prev_year),
+            days_in_prev_year: Self::days_in_provided_year(prev_year),
+            next_year: Self::year_as_hebrew(next_year),
+        }
+    }
+}
+
+impl Hebrew {
+    /// Constructs a new Hebrew Calendar
+    pub fn new() -> Self {
+        Self
+    }
+
+    // Hebrew New Moon
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2244
+    #[allow(dead_code)]
+    pub(crate) fn molad(h_year: i32, h_month: u8) -> Moment {
+        let y = if h_month < TISHRI { h_year + 1 } else { h_year };
+
+        let months_elapsed = (h_month as f64 - TISHRI as f64)
+            + (libm::floor((1.0 / 19.0) * (235.0 * y as f64 - 234.0)));
```

Comment:
Can you change the `(1/x) * somexpr` construction to `somexpr / x` ? The latter should give better precision (and it matches the book code, FWIW).

Reply:
done, thanks.